docker/docker-compose.mysql-nossl-server-ssession-connector_transparent.yml

```diff
@@ -0,0 +1,191 @@
+        volumes:
+            # Mount the whole ./.acrakeys directory to be able generate keys and
+            # place them in services' subdirectories
+            - ./.acrakeys:/keys
+        # Please specify ACRA_CLIENT_ID environment variable, otherwise run with
+        # default 'testclientid' client id
+        command: >-
+            --client_id=${ACRA_CLIENT_ID:-testclientid}
+            --generate_acraserver_keys
+            --keys_output_dir=/keys/acra-server
+            --keys_public_output_dir=/keys/acra-connector
+    # Create keys:
+    # - ./.acrakeys/acra-connector/${ACRA_CLIENT_ID}
+    # - ./.acrakeys/acra-server/${ACRA_CLIENT_ID}.pub
+    acra-keymaker_connector:
+        image: "cossacklabs/acra-keymaker:${ACRA_DOCKER_IMAGE_TAG:-latest}"
+        network_mode: "none"
+        environment:
+            ACRA_MASTER_KEY: ${ACRA_CONNECTOR_MASTER_KEY:-MElBVnhEeTd3b29JMFVVcnhGMXJPT3BxZUVwWW5wS3E=}
+        volumes:
+            - ./.acrakeys:/keys
+        command: >-
+            --client_id=${ACRA_CLIENT_ID:-testclientid}
+            --generate_acraconnector_keys
+            --keys_output_dir=/keys/acra-connector
+            --keys_public_output_dir=/keys/acra-server
+    # Create keys:
+    # - ./.acrakeys/acra-server/${ACRA_CLIENT_ID}_storage
+    # - ./.acrakeys/acra-writer/${ACRA_CLIENT_ID}_storage.pub
+    acra-keymaker_writer:
+        image: "cossacklabs/acra-keymaker:${ACRA_DOCKER_IMAGE_TAG:-latest}"
+        network_mode: "none"
+        environment:
+            ACRA_MASTER_KEY: ${ACRA_SERVER_MASTER_KEY:-N0MyZEVCRGY1cHA4ODRSVHp3c3BNeGtDekxaV3BhTHI=}
+        volumes:
+            - ./.acrakeys:/keys
+        command: >-
+            --client_id=${ACRA_CLIENT_ID:-testclientid}
+            --generate_acrawriter_keys
+            --keys_output_dir=/keys/acra-server
+            --keys_public_output_dir=/keys/acra-server
+
+    # Create file with accounts for HTTP access and key for decrypt it
+    # - ./.acrakeys/acra-server/httpauth.accounts
+    # - ./.acrakeys/acra-server/auth_key
+    acra-authmanager:
+        image: "cossacklabs/acra-authmanager:${ACRA_DOCKER_IMAGE_TAG:-latest}"
+        network_mode: "none"
+        environment:
+            ACRA_MASTER_KEY: ${ACRA_SERVER_MASTER_KEY:-N0MyZEVCRGY1cHA4ODRSVHp3c3BNeGtDekxaV3BhTHI=}
+        volumes:
+            - ./.acrakeys:/keys
+        command: >-
+            --set
+            --user=${ACRA_HTTPAUTH_USER:-test}
+            --password=${ACRA_HTTPAUTH_PASSWORD:-test}
+            --file=/keys/acra-server/httpauth.accounts
+            --keys_dir=/keys/acra-server/
+
+    # MySQL container
+    mysql:
+        image: mariadb:10.3
+        # INSECURE!!! You MUST define your own DB name and credentials
+        environment:
+            MYSQL_ROOT_PASSWORD: ${MYSQL_ROOT_PASSWORD:-hy53uBLmBuihUK29}
+            MYSQL_DATABASE: ${MYSQL_DATABASE:-test}
+            MYSQL_USER: ${MYSQL_USER:-test}
+            MYSQL_PASSWORD: ${MYSQL_PASSWORD:-test}
```

Comment:
Is there any good secure default that we can use so that we don't have insecure-by-default examples? Like, load the credentials from a file or whatever.

Also, why exactly is this insecure when having Acra master key in the same file is apparently secure? Well, in some other places we *do have* warnings stating that using our hardcoded and published master keys is also insecure.

I'd suggest to change these legitimately looking base64 strings into something prominent, like `NOT-FOR-PRODUCTION-PLEASE-CHANGE` for the avoidance of doubt.

Reply:
The primary goals of this set of docker demos are:
* quick start with one command
* show different behaviours
* show main features
* to be illustrative and understandable
* to show relations between components

We cannot make such easy one-command demos _really_ secure, because it is not only about one docker-compose configuration. It should be a set of approaches in this configuration, in the applciation, and in the infrastructure design.

But each step to make it a bit secure will make it much less readable and much more complex.

Apparently, secure infrastructure and application design is out of scope for this simple demos. So let's write with big red letters obvious warnings and will not became too dispersed.